Fix bug for issue number 3319 (Required files client side validation)

JSeifBY opened the pull request:
when adding a new post with a required field type 'File'; if i let the file empty the form is submitted and generate an error 'empty file' but the problem that the post has been added to the list without a file.

bug fix : iterate over pods required 'file' field type test if at least one file is selected, if not prevent submit and alert message showing 'You have same empty required files!'

sc0ttkclark commented:
I like what you're doing here, but we need a couple more tweaks to the text shown and I'd like to make the messaging / alert a little less `alert()`, maybe showing a message at the top of the screen `<div id="message" class="error">....</div>` or something like that, as well as messages and/or red markings on individual fields.

JSeifBY commented:
Sorry for that response delay,
i have just removed the alert dialog and added an inline error messages. I was thinking about adding notification manager like https://notifyjs.com/ to notify user of occurred errors (not only for required files), for example in long scroll pages we can't find the errors in a short time.

sc0ttkclark commented:
@JSeifBY can you provide collaborator access to your repository?

JSeifBY commented:
Done!

sc0ttkclark commented on ui/js/jquery.pods.js:
```diff
@@ -122,6 +120,26 @@
+							var $fieldName = $fieldRow.find( 'label' ).text();
+
+							// @todo Needs i18n work
+							$fieldRow.append(' <td class="pods-validate-error-message">' + $fieldName.replace('*', '' ) + ' is required.</td>' );
```
@JoryHogeveen Needs i18n JS inclusion here, can you add?

sc0ttkclark replied:
```suggestion
							$fieldRow.append(' <td class="pods-validate-error-message">' + PodsI18n.__( '%s is required.' ).replace( '%s', $fieldName.replace('*', '' ) ) + '</td>' );
```

sc0ttkclark commented:
After some changes, it looks better. We need to update some of the JS here though to make sure the markup is right for 2.7

@pglewis can you check the logic of this out and confirm if markup it's using to add/remove the message is correct? Feel free to suggest changes that would be better overall for per-field validation messages.

JoryHogeveen commented on ui/js/jquery.pods.js:
```diff
@@ -122,6 +120,26 @@
+							var $fieldName = $fieldRow.find( 'label' ).text();
+
+							// @todo Needs i18n work
+							$fieldRow.append(' <td class="pods-validate-error-message">' + $fieldName.replace('*', '' ) + ' is required.</td>' );
```
I can't edit the file, maybe new PR after this one?

sc0ttkclark commented:
@JSeifBY can you add access for @JoryHogeveen to your repo?

sc0ttkclark commented:
Also @pglewis

sc0ttkclark commented on ui/js/jquery.pods.js:
```diff
@@ -122,6 +120,26 @@
+
+							var $fieldName = $fieldRow.find( 'label' ).text();
+
+							// @todo Needs i18n work
```
```suggestion
```
